.github/actions/deploy-environment/action.yml

```diff
@@ -47,15 +47,48 @@ runs:
       run: |
         access_url=$(terraform -chdir=terraform/application output -json urls | jq -r '.[0]')
         echo "ACCESS_URL=$access_url" >> $GITHUB_OUTPUT
+        check_url=$(terraform -chdir=terraform/application output -json urls | jq -r '.[1]')
+        echo "CHECK_URL=$check_url" >> $GITHUB_OUTPUT
+
+        external_urls=$(terraform -chdir=terraform/application output -json external_urls)
+        external_access_url=$(echo $external_urls | jq -r '.[0]')
+        external_check_url=$(echo $external_urls | jq -r '.[1]')
+        echo "EXTERNAL_ACCESS_URL=$external_access_url" >> $GITHUB_OUTPUT
+        echo "EXTERNAL_CHECK_URL=$external_check_url" >> $GITHUB_OUTPUT
 
     - name: Run smoke tests
       shell: bash
       run: |
-        # Parse the JSON array of URLs into a Bash array
-        urls=$(echo "$HOSTNAMES" | jq -r '.[]')
+        # Parse the JSON array of internal URLs into a Bash array
+        internal_urls=$(terraform -chdir=terraform/application output -json urls | jq -r '.[]')
+
+        # Get the external URLs
+        external_access_url="${{ steps.set_outputs.outputs.EXTERNAL_ACCESS_URL }}"
+        external_check_url="${{ steps.set_outputs.outputs.EXTERNAL_CHECK_URL }}"
+
+        echo "Internal URLs:"
+        echo "$internal_urls"
 
-        # Loop over each URL and perform the curl check on the /health/all.json endpoint
-        for url in $urls; do
+        # Function to perform health check
+        perform_health_check() {
+          local url="$1"
+          echo "Check health for $url..."
```

Comment:
we should remove the $url check. We only want the $url/health/all.json check. The url check won't really work anyway as it's behind basic auth.